appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/node/ResourcePropertyNode.java

```diff
@@ -84,15 +84,14 @@ public Node writeDescriptor(Node node, Descriptor desc) {
             properties = ((JMSDestinationDefinitionDescriptor) desc).getProperties();
         }
 
-
-        Set keys = properties.keySet();
-
-        for (Object key : keys) {
-            String name = (String) key;
-            String value = (String) properties.get(name);
-            Node propertyNode = appendChild(node, TagNames.RESOURCE_PROPERTY);
-            appendTextChild(propertyNode, TagNames.RESOURCE_PROPERTY_NAME, name);
-            appendTextChild(propertyNode, TagNames.RESOURCE_PROPERTY_VALUE, value);
+        if (properties != null) {
```

Comment:
There seems to be a set of classes we are able to extract `properties` from. Should this be called with another implementation class for `desc` we don't know how to access properties so we have to skip adding text nodes. Behaviour change: NPE for certain - maybe plainly theoretical - descriptor types becomes NOOP.

I did not add any logging or throwing of an exception as this seems to do an extra thing for a known set of classes. In such a case I would expect unknown classes just not to have this extra functionality.

Reply:
I believe there is set of xml schema checks before these nodes are filled, so it is quite possible that this is pure theory (but you can try creating a descriptor that would trigger this ;)